smoke-tests/src/test/java/io/opentelemetry/smoketest/AppServerTestRunner.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.smoketest;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+import org.spockframework.runtime.Sputnik;
+
+/**
+ * Customized spock test runner that runs tests on multiple app server versions based on {@link
+ * AppServer} annotations. This runner selects first server based on {@link AppServer} annotation
+ * calls setupSpec, all test method and cleanupSpec, selects next {@link AppServer} and calls the
+ * same methods. This process is repeated until tests have run for all {@link AppServer}
+ * annotations. Tests should start server in setupSpec and stop it in cleanupSpec.
+ */
+public class AppServerTestRunner extends Sputnik {
+  private static final Map<Class<?>, AppServer> runningAppServer =
+      Collections.synchronizedMap(new HashMap<>());
```

Comment:
can this be `ThreadLocal<AppServer> currentAppServer`?

Reply:
I didn't use ThreadLocal because of https://github.com/spockframework/spock/pull/67 Apparently there is a way to run tests in parallel (which I believe we don't use). Using a Map seemed like a safer choice.